File: strings/1239.cpp
#include <iostream>

void processar_string(std::string s)
{
  bool ngr = false;
  bool itl = false;

  for (size_t i = 0; i < s.length(); ++i) {
    if (s[i] == '_') {
      std::cout << (itl? "</i>" : "<i>");
      itl = !itl;
    } else if (s[i] == '*') {
      std::cout << (ngr? "</b>" : "<b>");
      ngr = !ngr;
    } else
      std::cout << s[i];
  }
  std::cout << "\n";
}

int main(void)
{
  std::string linha;

  std::cin.tie(NULL);
  std::cin.sync_with_stdio(false);

  while (std::getline(std::cin, linha))
    processar_string(linha);

  return 0;
}

File: strings/1239.py
def processar_string(s):
  i = 0
  italico = False
  negrito = False

  while i < len(s):
    if s[i] == '_':
      if italico:
        print('</i>', end='')
        italico = False
      else:
        print('<i>', end='')
        italico = True
    elif s[i] == '*':
      if negrito:
        print('</b>', end='')
        negrito = False
      else:
        print('<b>', end='')
        negrito = True
    else:
      print(s[i], end='')
    i += 1
  print()

while True:
  try:
    linha = input()
    processar_string(linha)
  except:
    break
